mypy is failing

```
Run mypy --ignore-missing-imports .
  mypy --ignore-missing-imports .
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.7.10/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.7.10/x64/lib
tests/udf/test_find.py:15: error: Incompatible types in assignment (expression has type "MarkDecorator", variable has type "List[MarkDecorator]")
ibis_bigquery/udf/__init__.py:18: error: Need type annotation for '_udf_name_cache'
tests/udf/test_core.py:16: error: Incompatible types in assignment (expression has type "MarkDecorator", variable has type "List[MarkDecorator]")
ibis_bigquery/client.py:365: error: Item "None" of "Optional[str]" has no attribute "split"
tests/udf/test_udf_execute.py:20: error: Incompatible types in assignment (expression has type "MarkDecorator", variable has type "List[MarkDecorator]")
Found 5 errors in 5 files (checked 19 source files)
Error: Process completed with exit code 1.
```

--- a/ibis_bigquery/client.py
+++ b/ibis_bigquery/client.py
@@ -360,12 +360,11 @@ def parse_project_and_dataset(
     None
 
     """
-    try:
-        data_project, dataset = dataset.split('.')
-    except (ValueError, AttributeError):
-        billing_project = data_project = project
-    else:
+    if dataset is not None and "." in dataset:
+        data_project, dataset = dataset.split(".")
         billing_project = project
+    else:
+        billing_project = data_project = project
 
     return data_project, billing_project, dataset
 

--- a/ibis_bigquery/udf/__init__.py
+++ b/ibis_bigquery/udf/__init__.py
@@ -2,6 +2,7 @@ import collections
 import functools
 import inspect
 import itertools
+from typing import Dict, Iterable
 
 import ibis.expr.datatypes as dt
 import ibis.expr.rules as rlz
@@ -15,7 +16,7 @@ from ..udf.core import PythonToJavaScriptTranslator
 
 __all__ = ('udf',)
 
-_udf_name_cache = collections.defaultdict(itertools.count)
+_udf_name_cache : Dict[str, Iterable[int]] = collections.defaultdict(itertools.count)
 
 
 def create_udf_node(name, fields):

--- a/tests/udf/test_core.py
+++ b/tests/udf/test_core.py
@@ -7,13 +7,6 @@ from ibis.compat import PY38
 
 from ibis_bigquery.udf.core import PythonToJavaScriptTranslator, SymbolTable
 
-if PY38:
-    # ref: https://github.com/ibis-project/ibis/issues/2098
-    # note: UDF is already skipt on CI
-    pytestmark = [pytest.mark.bigquery, pytest.mark.udf]
-else:
-    pytestmark = pytest.mark.bigquery
-
 
 def test_symbol_table():
     symbols = SymbolTable()

--- a/tests/udf/test_find.py
+++ b/tests/udf/test_find.py
@@ -6,13 +6,6 @@ from ibis.util import is_iterable
 
 from ibis_bigquery.udf.find import find_names
 
-if PY38:
-    # ref: https://github.com/ibis-project/ibis/issues/2098
-    # note: UDF is already skipt on CI
-    pytestmark = [pytest.mark.bigquery, pytest.mark.udf]
-else:
-    pytestmark = pytest.mark.bigquery
-
 
 def parse_expr(expr):
     body = parse_stmt(expr)

--- a/tests/udf/test_udf_execute.py
+++ b/tests/udf/test_udf_execute.py
@@ -11,14 +11,6 @@ from pytest import param
 import ibis_bigquery
 from ibis_bigquery import udf  # noqa: E402
 
-if PY38:
-    # ref: https://github.com/ibis-project/ibis/issues/2098
-    # note: UDF is already skipt on CI
-    pytestmark = [pytest.mark.bigquery, pytest.mark.udf]
-else:
-    pytestmark = pytest.mark.bigquery
-
-
 PROJECT_ID = os.environ.get('GOOGLE_BIGQUERY_PROJECT_ID', 'ibis-gbq')
 DATASET_ID = 'testing'
 

--- a/tests/unit/test_client.py
+++ b/tests/unit/test_client.py
@@ -1,0 +1,20 @@
+import pytest
+
+import ibis_bigquery.client
+
+
+@pytest.mark.parametrize(
+    ["project", "dataset", "expected"],
+    [
+        ("my-project", None, ("my-project", "my-project", None)),
+        ("my-project", "my_dataset", ("my-project", "my-project", "my_dataset")),
+        (
+            "billing-project",
+            "data-project.my_dataset",
+            ("data-project", "billing-project", "my_dataset"),
+        ),
+    ]
+)
+def test_parse_project_and_dataset(project, dataset, expected):
+    got = ibis_bigquery.client.parse_project_and_dataset(project, dataset)
+    assert got == expected

